### PR TITLE
feat: add option to ignore directories from coverage report

### PR DESCRIPTION
**Pull request by dimazhornyk**
## Motivation

Some files shouldn't be displayed in a coverage report, e.g. utility contracts or dev contracts, and currently there is no way to ignore them from the coverage report. There is a popular issue describing the problem https://github.com/foundry-rs/foundry/issues/2988.

## Solution

This PR introduces such a possibility using a cli argument `--no-coverage-path` or config key `no_coverage_path`.
On the implementation side, it filters out all the ignored files right before the report output. It can't be done earlier because there may be files dependent on the ignored ones, which will not be covered if we remove the ignored files earlier.

### CODE REVIEW

File: crates/evm/coverage/src/lib.rs [hunk 1]
```diff
@@ -151,6 +154,30 @@ impl CoverageReport {
         }
         Ok(())
     }
+
+    pub fn filter_out_ignored_sources(&mut self, filter: &impl CoverageFilter) {
```
[the remark this answers is not in the document]
Comment (dimazhornyk) [reply crates/evm/coverage/src/lib.rs:162]: added

File: crates/evm/coverage/src/lib.rs [hunk 2]
```diff
@@ -151,6 +154,30 @@ impl CoverageReport {
...
+                    Path::new(&self.get_source_path(version, item.loc.source_id))
+                ))
+                .cloned()
+                .collect();
+            new_items.insert(version.clone(), new_items_for_version);
+        }
+        self.items = new_items;
+    }
+
+    fn get_source_path(&self, version: &Version, source_id: usize) -> String {
```
[the remark this answers is not in the document]
Comment (dimazhornyk) [reply crates/evm/coverage/src/lib.rs:180]: added

File: crates/forge/bin/cmd/test/filter.rs [hunk 3]
```diff
@@ -40,6 +41,10 @@ pub struct FilterArgs {
         value_name = "GLOB"
     )]
     pub path_pattern_inverse: Option<GlobMatcher>,
+
+    /// Only show coverage for files that do not match the specified glob pattern.
+    #[arg(long = "no-coverage-path", visible_alias = "ncp", value_name = "GLOB")]
```
[the remark this answers is not in the document]
Comment (dimazhornyk) [reply crates/forge/bin/cmd/test/filter.rs]: renamed

**Comment by zerosnacks** [comment]
Hi @dimazhornyk thanks for your PR!

Looks like this is almost ready for review, would be great to get this merged

To resolve the formatting issue make sure to run: `cargo +nightly fmt --all`

**Comment by mattsse** [comment]
sorry about dropping this @dimazhornyk ...

if you don't have the capacity for fixing the conflicts, we can take it from here @zerosnacks

**Comment by zerosnacks** [comment]
Picking this up, merging into `zerosnacks/ignore-directories-coverage` so fixes can be applied and the commits of @dimazhornyk stay applied

**Comment by zerosnacks** [comment]
Closing this PR in favor of https://github.com/foundry-rs/foundry/pull/8321

I've included the commits of @dimazhornyk

**Pull request closed without merging**